### PR TITLE
PERF: Avoid Series constructor in DataFrame(dict(...), columns=)

### LINKED ISSUE #24368: DataFrame constructor with dict of series misbehaving when `columns` specified
We take a strange path...

```diff
diff --git a/pandas/core/internals/construction.py b/pandas/core/internals/construction.py
index c43745679..54108ec58 100644
--- a/pandas/core/internals/construction.py
+++ b/pandas/core/internals/construction.py
@@ -552,6 +552,7 @@ def sanitize_array(data, index, dtype=None, copy=False,
             data = data.copy()
 
     # GH#846
+    import pdb; pdb.set_trace()
     if isinstance(data, (np.ndarray, Index, ABCSeries)):
 
         if dtype is not None:
```

```pdb
In [1]: import pandas as pd; import numpy as np

In [2]: a = pd.Series(np.array([1, 2, 3]))
> /Users/taugspurger/sandbox/pandas/pandas/core/internals/construction.py(556)sanitize_array()
-> if isinstance(data, (np.ndarray, Index, ABCSeries)):
(Pdb) c

In [3]: r = pd.DataFrame({"A": a, "B": a}, columns=['A', 'B'])
> /Users/taugspurger/sandbox/pandas/pandas/core/internals/construction.py(556)sanitize_array()
-> if isinstance(data, (np.ndarray, Index, ABCSeries)):
(Pdb) up
> /Users/taugspurger/sandbox/pandas/pandas/core/series.py(259)__init__()
-> raise_cast_failure=True)
(Pdb)
> /Users/taugspurger/sandbox/pandas/pandas/core/series.py(301)_init_dict()
-> s = Series(values, index=keys, dtype=dtype)
(Pdb)
> /Users/taugspurger/sandbox/pandas/pandas/core/series.py(204)__init__()
-> data, index = self._init_dict(data, index, dtype)
(Pdb)
> /Users/taugspurger/sandbox/pandas/pandas/core/internals/construction.py(176)init_dict()
-> arrays = Series(data, index=columns, dtype=object)
(Pdb)
> /Users/taugspurger/sandbox/pandas/pandas/core/frame.py(387)__init__()
-> mgr = init_dict(data, index, columns, dtype=dtype)
(Pdb) data
{'A': 0    1
1    2
2    3
dtype: int64, 'B': 0    1
1    2
2    3
dtype: int64}
(Pdb) d
> /Users/taugspurger/sandbox/pandas/pandas/core/internals/construction.py(176)init_dict()
-> arrays = Series(data, index=columns, dtype=object)
(Pdb) l
171         Segregate Series based on type and coerce into matrices.
172         Needs to handle a lot of exceptional cases.
173         """
174         if columns is not None:
175             from pandas.core.series import Series
176  ->         arrays = Series(data, index=columns, dtype=object)
177             data_names = arrays.index
178
179             missing = arrays.isnull()
180             if index is None:
181                 # GH10856
(Pdb) data
{'A': 0    1
1    2
2    3
dtype: int64, 'B': 0    1
1    2
2    3
dtype: int64}
```

I'm guessing we don't want to be passing a dict of Series into the Series constructor there

### LINKED ISSUE #25887: Empty dataframe creation
Hello,

Creating an empty dataframe with specified columns
```python
pd.__version__
# '0.24.2'

%timeit pd.DataFrame([], columns=['a'])
# 1.57 ms ± 20.9 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```
is way slower than it should be:
```python
%timeit pd.DataFrame([])
# 184 µs ± 1.68 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

%timeit pd.DataFrame([[1]], columns=['a'])
# 343 µs ± 3.33 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

### PR DESCRIPTION
**Pull request by mroeschke**
- [ ] closes #24368 (Replace xxxx with the GitHub issue number)
- [ ] closed #25887
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

```python
import pandas as pd, numpy as np
a = pd.Series(np.array([1, 2, 3]))
columns = range(10_000)
data = {i: a for i in columns}
%timeit pd.DataFrame(data, columns=columns)

51.3 ms ± 96 µs per loop (mean ± std. dev. of 7 runs, 10 loops each) # PR
102 ms ± 1.8 ms per loop (mean ± std. dev. of 7 runs, 10 loops each) # main
```

```python
In [1]: import pandas as pd

In [2]: %timeit pd.DataFrame([], columns=['a'])
37.3 µs ± 231 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)  # PR
183 µs ± 584 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)  # main
```

### CODE REVIEW

**Comment by mroeschke** [comment]
When you have a chance, could you take a look @jbrockmendel

**Comment by jbrockmendel** [comment]
will take a look.  i like the idea of getting Series out of here

**Comment by jbrockmendel** [comment]
LGTM

File: pandas/core/internals/construction.py [hunk 1]
```diff
@@ -354,45 +356,56 @@ def dict_to_mgr(
...
+
+        for i, column in enumerate(columns):
+            try:
+                idx = data_keys.get_loc(column)
+            except KeyError:
+                midxs.add(i)
+                continue
+            array = data_values[idx]
+            arrays[i] = array
+            if is_scalar(array) and isna(array):
```
[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/core/internals/construction.py:376]: Correct, doesn't look like we have one explicitly so I'll add one

**Pull request merged**